### PR TITLE
BPTL Dashboard: UI for home collection (Part 1)

### PR DESCRIPTION
**Pull request by abhinavjonnada82**
This PR addresses following feature:
https://github.com/episphere/connect/issues/771

Related PR: https://github.com/episphere/connectFaas/pull/448

Checklist:
- [X] Code cleanup
- [X] Check for ES Lint warnings
- [ ] Verify test cases
- [X] Check for GIT conflicts 
- [ ] Verified unit tests pass with current changes [N/A]
- [ ] Any dependencies or modules  [N/A]
- [X] Attach PoC
- [ ] Add notes

PoC:

- Assemble Kit:
Screenshot 2023-10-31 at 10 08 57 AM

- Print Labels:
Screenshot 2023-10-31 at 10 11 18 AM

Notes:

### CODE REVIEW

File: src/fieldToConceptIdMapping.js [hunk 1]
```diff
@@ -286,11 +286,22 @@ export default {
...
+
+    // home collection
+    supplyKitTrackingNum: 531858099,
+    supplyKitId: 690210658,
+    returnKitId: 194252513,
+    collectionCupId: 259846815,
+    collectionCardId: 786397882,
+    UKID: 687158491,
+    kitType: 379252329
+
```
Comment (jhflorey) [inline src/fieldToConceptIdMapping.js:303]: NIT: SHould we remove space line

File: src/pages/homeCollection/printLabels.js [hunk 2]
```diff
@@ -0,0 +1,138 @@
+import { homeCollectionNavbar } from "./homeCollectionNavbar.js";
+import { getIdToken, showAnimation, hideAnimation, baseAPI, appState } from "../../shared.js";
+import { nonUserNavBar, unAuthorizedUser } from "./../../navbar.js";
```
Comment (jhflorey) [inline src/pages/homeCollection/printLabels.js:3]: `unAuthorizedUser` appears to be unused

File: src/pages/homeCollection/printLabels.js [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+import { activeHomeCollectionNavbar } from "./activeHomeCollectionNavbar.js";
+
+
+const contentBody = document.getElementById("contentBody");
+
+export const printLabelsScreen = async (auth, route) => {
```
Comment (jhflorey) [inline src/pages/homeCollection/printLabels.js:9]: `route` appears be unused

File: src/pages/homeCollection/printLabels.js [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+            if (arrayLengthToProcess >= numberToPrint) {
+              const arrayToProcess = appState.getState().totalAddresses
+              const remainingArrayToProcess = arrayToProcess.slice(numberToPrint, appState.getState().length)
+              appState.setState({totalAddresses: remainingArrayToProcess })
+              appState.setState({'length': remainingArrayToProcess.length })
+              generateParticipantCsv(arrayToProcess.slice(0, numberToPrint));
+              printLabelsTemplate(name);
+                // Display success message
+              const alertList = document.getElementById('alert_placeholder');
+              const template = `
```
Comment (jhflorey) [inline src/pages/homeCollection/printLabels.js:95]: NIT: Should we assign the string value for alertList.innerHTML without declaring `template`

**Review by jhflorey — APPROVED** [review]

File: src/pages/homeCollection/kitAssembly.js [hunk 5]
```diff
@@ -1,159 +1,23 @@
...
-    alertTemplate(message, status);
-  } else {
-    let message = "The kit was not saved successfully!";
-    let status = "warn";
-    alertTemplate(message, status);
-  }
-};
-
-const kitAssemblyTemplate = async (user, name, auth, route) => {
+const kitAssemblyTemplate = async (name, auth, route) => {
```
Comment (Gbarra9) [inline src/pages/homeCollection/kitAssembly.js:20]: `auth` and `route` parameters are not being used in `kitAssemblyTemplate`

File: src/pages/homeCollection/kitAssembly.js [hunk 6]
```diff
@@ -162,667 +26,199 @@ const kitAssemblyTemplate = async (user, name, auth, route) => {
...
+    event.preventDefault();
+    if (event.key === 'Enter') {
+        document.getElementById("saveKit").click();
     }
+});
+}
 
-    tableBody.innerHTML = tableRow;
+const processAssembledKit = () => {
+  const a = document.getElementById('saveKit');
```
Comment (Gbarra9) [inline src/pages/homeCollection/kitAssembly.js:105]: We could rename `a` to `saveKitButton` or another variable name to be more descriptive.

File: src/pages/homeCollection/kitAssembly.js [hunk 7]
```diff
@@ -162,667 +26,199 @@ const kitAssemblyTemplate = async (user, name, auth, route) => {
...
-      );
-      supplyKitInput.style.borderColor = "#E00000";
-      inputSupplyKitErrorMessage.innerHTML = `Supply Kit ID length must be 9 characters. <br/>The format must start with an all uppercase CON, followed by 6 digits, each digit can be a number from 0 to 9.`;
-    }
-    return;
-  });
+const storeAssembledKit = async (kitData) => {
+  const idToken = await getIdToken();
+  let api = `addKitData`
+  if(appState.getState().UKID !== ``) {
```
Comment (Gbarra9) [inline src/pages/homeCollection/kitAssembly.js:173]:
```suggestion
  if (appState.getState().UKID !== ``) { 
```
NIT: spacing

File: src/pages/homeCollection/kitAssembly.js [hunk 8]
```diff
@@ -162,667 +26,199 @@ const kitAssemblyTemplate = async (user, name, auth, route) => {
...
-        "color:#E00000;display:inline-block;font-size:.8rem;"
-      );
-      collectionCupInput.style.borderColor = "#E00000";
-      inputCollectionCupErrorMessage.innerHTML = `Collection Cup ID length must be 14 characters. <br/>The format must start with an all uppercase CX, followed by any capital letter from A-Z, followed by 6 digits, each digit can be a number from 0 to 9, followed by a space, and followed by 4 digits, each digit can be a number from 0 to 9. `;
+  if (response.status === 200) {
+    alertTemplate(`Kit saved successfully!`, `success`);
+    const existingKitData = JSON.parse(localStorage.getItem('tmpKitData'));
+    existingKitData.push(kitData);
+
+    if(appState.getState().UKID !== ``) {
```
Comment (Gbarra9) [inline src/pages/homeCollection/kitAssembly.js:192]:
```suggestion
    if (appState.getState().UKID !== ``) {
```
NIT: spacing

File: src/pages/homeCollection/kitAssembly.js [hunk 9]
```diff
@@ -854,93 +250,16 @@ const closeAlert = (status = "warn", duration = 5000) => {
...
-
-const matchCollectionCupCardClearRedBorder = (
-  collectionCupInputElement,
-  collectionCardInputElement
-) => {
-  collectionCupInputElement.style.borderColor = "";
-  collectionCardInputElement.style.borderColor = "";
-};
+const dropdownTrigger = (sitekeyName) => {
+  let a = document.getElementById('dropdownSites');
```
Comment (Gbarra9) [inline src/pages/homeCollection/kitAssembly.js:254]: We can change this variable name `a` to another descriptive name.

**Review by Gbarra9 — APPROVED** [review]
Looks good. Left comments.

**Review by jhflorey — APPROVED** [review]

File: src/shared.js [hunk 10]
```diff
@@ -43,7 +43,7 @@ let api = '';
 
 if(location.host === urls.prod) api = 'https://api-myconnect.cancer.gov/biospecimen?';
 else if(location.host === urls.stage) api = 'https://api-myconnect-stage.cancer.gov/biospecimen?';
-else api = 'https://us-central1-nih-nci-dceg-connect-dev.cloudfunctions.net/biospecimen?';
+else api = 'http://127.0.0.1:5001/nih-nci-dceg-connect-dev/us-central1/biospecimen?';
```
Comment (Gbarra9) [inline src/shared.js:46]: I want to confirm if this localhost url is correct?

**Review by Gbarra9 — APPROVED** [review]

File: src/pages/homeCollection/kitAssembly.js [hunk 11]
```diff
@@ -1,828 +1,325 @@
...
+        const responseStoredStatus = await storeAssembledKit(obj);
+        if (responseStoredStatus) {
+          document.getElementById('scannedBarcode').value = ``
+          document.getElementById('supplyKitId').value = ``
+          document.getElementById('returnKitId').value = ``
+          document.getElementById('cupId').value = ``
+          document.getElementById('cardId').value = ``
+        }
+        else {
+          document.getElementById('scannedBarcode').value = scannedBarcodeValue
```
Comment (anthonypetersen) [inline src/pages/homeCollection/kitAssembly.js:180]: are these lines needed? would the values not already be set?

File: src/pages/homeCollection/kitAssembly.js [hunk 12]
```diff
@@ -1,828 +1,325 @@
...
+    let api = `addKitData`
+    if (appState.getState().UKID !== ``) { 
+      api = `updateKitData` 
+      kitData[conceptIds.UKID] = appState.getState().UKID
     }
-    return;
-  });
+    const response = await fetch(`${baseAPI}api=${api}`, {
+      method: "POST",
+      body: JSON.stringify(kitData),
```
Comment (anthonypetersen) [inline src/pages/homeCollection/kitAssembly.js:258]: should this be in a function?

**Review by anthonypetersen — CHANGES_REQUESTED** [review]

**Review by anthonypetersen — APPROVED** [review]

**Pull request merged**